tests/upgrades/test_repository.py

```diff
@@ -203,3 +203,82 @@ def test_post_scenario_custom_repo_check(self, target_sat, pre_upgrade_data):
         )[0]
         result = rhel_client.execute(f'yum install -y {FAKE_4_CUSTOM_PACKAGE_NAME}')
         assert result.status == 0
+
+    @pytest.mark.pre_upgrade
+    def test_pre_scenario_custom_repo_check_sca_toggle(
+        self, target_sat, sat_upgrade_chost, save_test_data, default_location
+    ):
+        """This is a pre-upgrade scenario test to verify that repositories in a non-sca org
+        set to "Enabled" should be overridden to "Enabled(Override)" when upgrading to 6.14.
+
+        :id: preupgrade-65e1e312-a743-4605-b226-f580f523377f
+
+        :steps:
+            1. Before Satellite upgrade.
+            2. Create new Organization, Location.
+            3. Create Product, Custom Repository, Content view.
+            4. Create Activation Key and add Subscription.
+            5. Create a Content Host, register it, and check Repository Sets for enabled Repository.
+
+        :expectedresults:
+
+            1. Custom Repository is created.
+            2. Custom Repository is enabled on Host.
+
+        """
+        org = target_sat.api.Organization().create()
```

Comment:
You could use function_org fixture and pass it down to the post via save_test_dat

Reply:
Even we could simply use module_org if we are not conflicting in these 2 scenaeios in pre_upgrade part.

Reply:
for this scenario, I want to make sure everything is new and no other tests conflict with it. It might pull more data into product details that is not needed.

Reply:
Function org fixture does the same thing that you are doing here , it saves u that line. Also if that fixture is being asked by other test it will create new org.